src/Tests/Microsoft.NET.TestFramework/TestAssetsManager.cs

```diff
@@ -109,13 +111,13 @@ public string GetAndValidateTestProjectDirectory(string testProjectName, string
 
         public static string GetTestDestinationDirectoryPath(
             string testProjectName,
-            string callingMethod,
+            string callingMethodAndFileName,
             string identifier)
```

Comment:
I think we could simplify this by just having a single `identifier` that combines `testProjectName`, `callingMethodAndFileName` and `identifier`. With the current setup the directory name will always be longer than 24 chars anyway so we'll always do the hashing.

That'd allow us to pass in the full path to the file name which would address @danmoseley's point about having the same file name with same test names in different directories.

Reply:
that is pretty much what I am doing. Note, on mac and linux, we do not hash them, so we don't want the folder name to be super long (it is hard for dev to debug).